### DISCUSSION TITLE
Contours not coloring correctly

### DISCUSSION THREAD

**Discussion by Plantain** [observablehq/plot#2230] [Q&A] [answered]
I am trying to draw contours, but the contours are not getting colored according to the specified domain. My expectation is that if I clamp the domain to the threshold range, that I would get the full range of colors used in the contouring, but that seems to not be the case for reasons I do not understand.

https://observablehq.com/d/79ecd1f124a6ec5e

I also do not understand the significance of Plot.identity from the docs (https://observablehq.com/plot/marks/contour) or understand when and how it should be used. It seems to just produce blank charts.

**Comment by Fil** [comment, marked as answer]
Plot.identity would color with the datum, which is an object `{time, altitude, ti}`. I think you want `fill: "ti"` instead. I'd also recommend to use `x: {type: "utc"}` (or `x: {type: "time"}`) for a more readable axis.

**Reply by Plantain** [reply]
Thanks!
